pkg/destroy/openstack/openstack.go

```diff
@@ -653,6 +653,57 @@ func getRouterByPort(client *gophercloud.ServiceClient, allPorts []ports.Port) (
 	return "", nil
 }
 
+func deleteLeftoverLoadBalancers(opts *clientconfig.ClientOpts, logger logrus.FieldLogger, networkID string) (bool, error) {
+	conn, err := clientconfig.NewServiceClient("load-balancer", opts)
+	if err != nil {
+		// Ignore the error if Octavia is not available for the cloud
+		var gerr *gophercloud.ErrEndpointNotFound
+		if errors.As(err, &gerr) {
+			logger.Debug("Skip load balancer deletion because Octavia endpoint is not found")
+			return true, nil
+		}
+		logger.Error(err)
+		return false, nil
+	}
+
+	listOpts := loadbalancers.ListOpts{
+		VipNetworkID: networkID,
+	}
+	allPages, err := loadbalancers.List(conn, listOpts).AllPages()
+	if err != nil {
+		logger.Error(err)
+		return false, nil
+	}
+
+	allLoadBalancers, err := loadbalancers.ExtractLoadBalancers(allPages)
+	if err != nil {
+		logger.Error(err)
+		return false, nil
+	}
+	deleteOpts := loadbalancers.DeleteOpts{
+		Cascade: true,
+	}
+	for _, loadbalancer := range allLoadBalancers {
+		if !strings.HasPrefix(loadbalancer.Description, "Kubernetes external service") {
+			logger.Debugf("Not deleting LoadBalancer %q with description %q", loadbalancer.ID, loadbalancer.Description)
+			continue
+		}
+		logger.Debugf("Deleting LoadBalancer %q", loadbalancer.ID)
+		err = loadbalancers.Delete(conn, loadbalancer.ID, deleteOpts).ExtractErr()
+		if err != nil {
+			// Ignore the error if the load balancer cannot be found
+			var gerr gophercloud.ErrDefault404
```

Comment:
please compare to the line #660

Reply:
Actually this is correct... This is how errors in gophercloud are organized